src/OpenTracing/StartSpanOptions.cs

```diff
@@ -0,0 +1,28 @@
+using System;
+using System.Collections.Generic;
+
+namespace OpenTracing
+{
+    /// <summary>
+    /// Provides a way to configure the behavior of <see cref="ITracer.StartSpan"/>.
+    /// </summary>
+    public class StartSpanOptions
```

Comment:
I don't think you should compare SpanBuilder to its Java version. In Java the way to create a span is through the buildSpan method. 
Here the method on the tracer interface is the StartSpan(OperationName, StartSpanOptions) method. This makes it very clear for the implementor. For the user this felt cumbersome if you have many start options. SpanBuilder is a helper extension for building the StartSpan parameters in a fluent API. There is no logic i the SpanBuilder. 
There are comments above that it could be postponed until we know more about the API usage, or extracted to an external extension library, but I feel that it is very simple and simplifies the user code and could be left here for now..